src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerGenotypingEngine.java

```diff
@@ -256,6 +295,79 @@ public CalledHaplotypes assignGenotypeLikelihoods(final List<Haplotype> haplotyp
     }
 
     /**
+     * Trim excessive alt alleles when the combination of ploidy and alleles is so large that exhaustively enumerating
+     * all possible genotypes is impractical.
+     * Use the highest haplotype score of each allele to decide which ones to trim away, and in case of tie, use the second best.
+     * @param alleleMapper          original allele to haplotype map to be trimmed
+     * @param desiredNumOfAlleles   desired allele count (ref allele won't be removed)
+     * @return trimmed map of the desired size
+     */
+    @VisibleForTesting
+    static Map<Allele, List<Haplotype>> reduceNumberOfAlternativeAllelesBasedOnHaplotypesScores(final Map<Allele, List<Haplotype>> alleleMapper, final int desiredNumOfAlleles) {
+
+        final PriorityQueue<AlleleScoredByHaplotypeScores> altAlleleMaxPriorityQ = new PriorityQueue<>((sa1, sa2) -> sa2.compareTo(sa1));
```

Comment:
Oh, I see, you didn't make AlleleScoredByHaplotypeScores `Comparable`.  Either make it Comparable or just use the comparator directly here.